### ISSUE TITLE
Nadam optimizer arguments description missed schedule_decay

### ISSUE THREAD

**Issue by slavikkom** [keras-team/keras#12189]
Both documentation https://keras.io/optimizers/ and the source code https://github.com/keras-team/keras/blob/master/keras/optimizers.py#L605 missed the description of an argument schedule_decay. Currently, it is:  
# Arguments
        lr: float >= 0. Learning rate.
        beta_1/beta_2: floats, 0 < beta < 1. Generally close to 1.
        epsilon: float >= 0. Fuzz factor. If `None`, defaults to `K.epsilon()`.
While actual list of parameters is: lr=0.002, beta_1=0.9, beta_2=0.999, epsilon=None, schedule_decay=0.004
I propose to add a line like schedule_decay: float >= 0, determines "what it actually does". Could do it myself also.

**Comment by gabrieldemarmiesse** [comment]
Thanks for the report. It strange that our continuous integration system didn't pick up this anomaly. This is an issue. This line: https://github.com/keras-team/keras/blob/e59570ae26670f788d6c649191031e4a8824f955/tests/test_documentation.py#L115 should have triggered an error.

Fix for this test welcome.

**Comment by RaphaelMeudec** [comment]
@gabrieldemarmiesse The test passes because it only checks that methods have a proper docstring (see `inspect.ismethod` in [here](https://github.com/keras-team/keras/blob/e59570ae26670f788d6c649191031e4a8824f955/tests/test_documentation.py#L21)). However, `__init__` has the function type. I don't think we want to apply `handle_function` to the init as the docstring in at the class level. Would adding a match between class docstring and init arguments make sense ?

**Comment by gabrieldemarmiesse** [comment]
Yes it would make sense and should be enforced. Since we use it for the docs, the docstring should be correct. This issue shows it. PR welcome to enforce this.

**Comment by gabrieldemarmiesse** [comment]
Fixed in #12192
